pkg/lang/javascript/aws_runtime/_fs.ts

```diff
@@ -0,0 +1,163 @@
+const flatted = require('flatted')
+const _ = require('lodash')
+const path = require('path')
+import { Readable } from 'stream'
+import {
+    S3Client,
+    PutObjectCommand,
+    GetObjectCommand,
+    ListObjectsCommand,
+    ListObjectsCommandInput,
+    DeleteObjectCommand,
+    HeadObjectCommand,
+} from '@aws-sdk/client-s3'
+
+const payloadBucketPhysicalName = process.env.KLOTHO_S3_PREFIX + '{{.PayloadsBucketName}}'
+const targetRegion = process.env['AWS_TARGET_REGION']
+
+const userBucketPath = '/files'
+
+const s3Client = new S3Client({ region: targetRegion })
+
+const streamToString = (stream: Readable): Promise<string> =>
+    new Promise((resolve, reject) => {
+        const chunks: Uint8Array[] = []
+        stream.on('data', (chunk) => chunks.push(chunk))
+        stream.on('error', reject)
+        stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf8')))
+    })
+
+async function getCallParameters(paramKey, dispatcherMode) {
+    let isEmitter = dispatcherMode === 'emitter' ? true : false
+    try {
+        const bucketParams = {
+            Bucket: payloadBucketPhysicalName,
+            Key: paramKey,
+        }
+        const result = await s3Client.send(new GetObjectCommand(bucketParams))
+
+        let parameters = ''
+        if (result.Body) {
+            parameters = await streamToString(result.Body as Readable)
+        }
+
+        if (isEmitter && Array.isArray(parameters)) {
+            // Emitters only have 1 parameter - the runtime saves an array, so we
+            // normalize the parameter
+            parameters = _.get(parameters, '[0]')
```

Comment:
How is this different from just `parameters[0]`? I tested, and it looks like `parameters[0]` evaluates to undefined if the array is empty, so we don't need index-bounds protection.

Reply:
a lot of this was copy paste from the old implementation, going to try to remove some of the more odd things like this